test/HttpApiClientTest.ts

```diff
@@ -0,0 +1,116 @@
+import {HttpApiClient, HttpClient} from "../src/httpClient";
+
+describe("HttpApiClient", () => {
+    let httpApiClient: HttpApiClient;
+
+    it("should return a valid baseUrl after construction", () => {
+
+        let defaultBaseUrl: string;
+
+        let httpClient = new HttpClient();
+        httpClient.configure((cfg) => {
+            defaultBaseUrl = cfg.options.baseUrl;
+        });
+
+        httpApiClient = new HttpApiClient();
+        httpApiClient.configure((cfg) => {
+            expect(cfg.options.baseUrl).toEqual(defaultBaseUrl);
+        });
+
+        httpApiClient = new HttpApiClient();
+        expect(httpApiClient.baseUrl).toBeUndefined();
+
+        httpApiClient = new HttpApiClient("https://myapi.crazy-factory.com");
+        expect(httpApiClient.baseUrl).toEqual("https://myapi.crazy-factory.com");
+
+    });
+
+    it("should throw errors for incorrect parameters", () => {
+
+        expect(() => {
+            httpApiClient.request("asdf", "/users/1", undefined, undefined, undefined);
+        }).toThrowError();
+
+    });
+
+    it("should throw an error on incorrect GET parameters", () => {
+
+        expect(() => {
+            httpApiClient.request("GET", "/users/1", {id: 1}, undefined, undefined);
+        }).toThrowError();
+
+        expect(() => {
+            httpApiClient.request("GET", "/users/1", undefined, [{file: "iamafile"}], undefined);
+        }).toThrowError();
+
+    });
+
+    it("should throw an error on incorrect HEAD parameters", () => {
+
+        expect(() => {
+            httpApiClient.request("HEAD", "/users/1", {id: 1}, undefined, undefined);
+        }).toThrowError();
+
+        expect(() => {
+            httpApiClient.request("HEAD", "/users/1", undefined, [{file: "iamafile"}], undefined);
+        }).toThrowError();
+
+    });
+
+    it("should throw an error on incorrect DELETE parameters", () => {
+
+        expect(() => {
+            httpApiClient.request("DELETE", "/users/1", undefined, [{file: "iamafile"}], undefined);
+        }).toThrowError();
+
+    });
+
+    it("should return a valid promise from get()", () => {
```

Comment:
You may test this just once calling request() as all other functions will call request() anyway. You can skip the test for the individual methods.

Additionally: Make sure, that they actually call request() and return it's value in the end. You can achieve this by creating a httpApiObject and wrapping/replacing it's request() method. You can then save the return value before it's return to you by get()/post() etc. and compare the result with toBe()

Also: Expand this test to use addMiddleware() to return a Promise. There's no need to call a real URL here.